### PR TITLE
[Task-3] Created a few new components, storybook configured and stories created

### PR DESCRIPTION
**Pull request by PetkevichVladislav**
Install Storybook

Install and configure [Storybook](https://storybook.js.org/) by following guide from the official website. In the end you should have a "storybook" npm script. When running "npm run storybook" it should start Storybook and open it in your browser.

Write Storybook stories for existing components

Create a story file for every component you created in previous modules. You should have at least one story for each of the following components (in total minimum 3):

⚛️ Counter
⚛️ SearchForm
⚛️ GenreSelect
You can write more stories to cover several states of GenreSelect or SearchForm, but this is optional.

Implement other components

Based on [Figma prototype](https://www.figma.com/file/fKGjrOqR6nJe6LYJopGCZ8/CDP-Home-Task-%E2%80%93-React-v1?node-id=0%3A2) implement the following components:

⚛️ MovieTile

This component should render a movie tile from the list of movies.

It should take properties to receive image url, movie name, release year, and a list of relevant genres. Alternatively, you can define one component property to take an object with all movie info. Additionally, the component should receive a callback property to capture click event.

Optionally, you can implement a context menu popup that opens when a user clicks on three dots button and contains "Edit" and "Delete" menu items.

⚛️ MovieDetails

This component will render movie details when a movie is selected from the list (clicked). The details include movie poster image on the left and the rest of info on the right.

The component should take properties to receive image url, movie name, release year, rating, duration and a description. Alternatively, you can specify a single property that accepts an object with all movie info.

⚛️ SortControl

This component will render a label "Sort by" and a select control to the right. Select should have the following options:

Release Date
Title
The component should take a property that specifies current selection. Additionally, it should take a callback property to handle selection changes. The callback should be called every time a user changes "Sort by" value. The new value should be passed in callback arguments.

Use Storybook when implementing every individual component. Write a story for every component you create. It will help you to check the result and play around with some interactivity before you embed components into the app.

Although, it's not required to implement high-fidelity design as per design prototype, it still makes sense to apply some styling to your components, so that you practice in styling React components and your final result looks good.

Write tests

Cover new components with tests using jest and @testing-library/react. Verify necessary data is rendered as well as that behavior works correctly. You can write snapshot tests to cover the rendering. Although, remember that snapshot tests are fragile and will fail any time you change your component markup.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,16 +1,32 @@
...
+    duration: "2h 34min",
+    description: "Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.—Soumitra"
+  });
+
   return (
     <div className="app">
-      <Header/>
-      <Main/>
-      <Footer/>
+      {/* <Header/> */}
```
Comment (iZamza) [inline src/App.tsx:25]: remove?

File: src/components/List/List.scss [hunk 2]
```diff
@@ -26,4 +29,49 @@
...
+
+        &:hover {
+            color: #F65261;
+
+            &::after {
+                background-color: #F65261;
+            }
+        }
+    }
+
```
Comment (iZamza) [inline src/components/List/List.scss:45]: please keep same spaces

File: src/components/List/List.tsx [hunk 3]
```diff
@@ -23,8 +24,12 @@ export const List: FC<IGenreListProps> = ({ itemNames, preselectedItemName, onSe
         onSelect(itemName);
     }
 
+    const listClassNames : string[] = ["list"];
+    if(selectedItemColor != null) {
```
Comment (iZamza) [inline src/components/List/List.tsx:28]: `!==`

File: src/components/List/List.tsx [hunk 4]
```diff
@@ -23,8 +24,12 @@ export const List: FC<IGenreListProps> = ({ itemNames, preselectedItemName, onSe
         onSelect(itemName);
     }
 
+    const listClassNames : string[] = ["list"];
```
Comment (iZamza) [inline src/components/List/List.tsx:27]: probably better to use `useState`

File: src/components/Menu/Menu.tsx [hunk 5]
```diff
@@ -2,23 +2,21 @@ import "./Menu.scss"
...
 import { useCallback, useState } from "react";
+import { SortControl, SortOption } from "../SortControl/SortControl";
 
 const Menu = () => {
-    const [itemNames, setItemNames] = useState<string[]>(["Action", "Adventure", "Comedy", "Drama", "Sci-Fi", "Thriller", "Western"]);
+    const [itemNames, setItemNames] = useState<string[]>(["Action", "Comedy", "Drama", "Sci-Fi", "Thriller", "Western"]);
     const [preselectedItemName, setSelectedItemName] = useState<string>("Sci-Fi");
-    const onSelect = useCallback((query: string) => { console.log(query); }, [])
+    const onSelect = useCallback((query: string) => { console.log(query); }, []);
+    const onSortOptionChange = useCallback((currentOption:SortOption) => console.log(currentOption), []);
```
Comment (iZamza) [inline src/components/Menu/Menu.tsx:11]: looks  like this function doesn't work

File: src/components/MovieCard/MovieCard.tsx [hunk 6]
```diff
@@ -1,17 +1,71 @@
...
+        },
+        {
+            onClick: () => console.log('Delete'),
+            menuItem: 'Delete',
+        }
+    ]);
+
+    const handleClosePopupMenu = () => {
+        setIsPopupMenuOpened(false);
+        console.log("close form");
```
Comment (iZamza) [inline src/components/MovieCard/MovieCard.tsx:31]: remove please

File: src/components/MovieCard/MovieCard.tsx [hunk 7]
```diff
@@ -1,17 +1,71 @@
...
+    imageUrl: string;
+    movieName: string;
+    releaseYear: string;
+    genres: string[];
+    onClick: () => void;
+}
+
+export const MovieCard: FC<IMovieCard> = (movie: IMovieCard) => {
+    const [isMouseHoveredOnCard, setIsMouseHoveredOnCard] = useState(false);
+    const [isPopupMenuOpened, setIsPopupMenuOpened] = useState(false);
```
Comment (iZamza) [inline src/components/MovieCard/MovieCard.tsx:17]: I know that it's not required, but use `useState<boolean>(false)`

File: src/components/MovieCard/MovieCard.tsx [hunk 8]
```diff
@@ -1,17 +1,71 @@
...
+        setIsMouseHoveredOnCard(false);
+    }
+
+    const handleCardClick = () => {
+        movie.onClick();
+    }
+
+    const handleMenuClick = (event: React.MouseEvent<HTMLSpanElement, MouseEvent>) => {
+        event.stopPropagation();
+        console.log("open form");
```
Comment (iZamza) [inline src/components/MovieCard/MovieCard.tsx:49]: remove please

File: src/components/MovieDetails/MovieDetails.tsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    movieName: string;
+    movieGenres: string[];
+    releaseYear: string;
+    rating: number;
+    duration: string;
+    description: string;
+}
+
+export const MovieDetails: FC<IMovieDetails> = (details: IMovieDetails) => {
+    let score = details.rating > 10 ? 10 : details.rating;
```
Comment (iZamza) [inline src/components/MovieDetails/MovieDetails.tsx:17]: is it magic numbers?

File: src/components/MovieGrid/MovieGrid.tsx [hunk 10]
```diff
@@ -3,22 +3,30 @@ import "./MovieGrid.scss"
...
-                <MovieCard/>
-                <MovieCard/>
-                <MovieCard/>
+                <MovieCard {...movieData} onClick={() => console.log("card clicked")}/>
+                <MovieCard {...movieData} onClick={() => console.log("card clicked")}/>
+                <MovieCard {...movieData} onClick={() => console.log("card clicked")}/>
+                <MovieCard {...movieData} onClick={() => console.log("card clicked")}/>
+                <MovieCard {...movieData} onClick={() => console.log("card clicked")}/>
+                <MovieCard {...movieData} onClick={() => console.log("card clicked")}/>
+                <MovieCard {...movieData} onClick={() => console.log("card clicked")}/>
```
Comment (iZamza) [inline src/components/MovieGrid/MovieGrid.tsx:29]: looks like loop? :)

File: src/stories/components/Menu.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+    parameters: {
+        docs: {
+            description: {
+                component: "Component renders menu."
+            }
+        }
+    }
+} as ComponentMeta<typeof Menu>
+
+const Template : ComponentStory<typeof Menu> = ( ) =>  <Menu/>;
```
Comment (iZamza) [inline src/stories/components/Menu.stories.tsx:17]: `()`

File: src/stories/components/MovieGrid.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+    parameters: {
+        docs: {
+            description: {
+                component: "Component renders component that shows movie cards."
+            }
+        }
+    }
+} as ComponentMeta<typeof MovieGrid>
+
+const Template : ComponentStory<typeof MovieGrid> = ( ) =>  <MovieGrid/>;
```
Comment (iZamza) [inline src/stories/components/MovieGrid.stories.tsx:16]: `<MovieGrid />`

**Review by iZamza — APPROVED** [review]
Good job

File: src/components/List/List.scss (hunk 2 above)
Comment (PetkevichVladislav) [reply src/components/List/List.scss:45]: redundant space removed, thank you for your comment.

File: src/components/List/List.tsx (hunk 4 above)
Comment (PetkevichVladislav) [reply src/components/List/List.tsx:27]: but it is not state of component, it's just class name

File: src/components/Menu/Menu.tsx (hunk 5 above)
Comment (PetkevichVladislav) [reply src/components/Menu/Menu.tsx:11]: It is work. To trigger this function, you need to click on any genre in menu. Then you can see the genre name in console 
image

File: src/components/MovieCard/MovieCard.tsx (hunk 6 above)
Comment (PetkevichVladislav) [reply src/components/MovieCard/MovieCard.tsx:31]: removed, thank you for your comment

File: src/components/MovieCard/MovieCard.tsx (hunk 7 above)
Comment (PetkevichVladislav) [reply src/components/MovieCard/MovieCard.tsx:17]: types specifications added, thank you for your comment

File: src/components/MovieDetails/MovieDetails.tsx (hunk 9 above)
Comment (PetkevichVladislav) [reply src/components/MovieDetails/MovieDetails.tsx:17]: Constant introduced for this numbers, thank you for your comment

File: src/components/MovieGrid/MovieGrid.tsx (hunk 10 above)
Comment (PetkevichVladislav) [reply src/components/MovieGrid/MovieGrid.tsx:29]: it was test data, replaced with array, thank you for your comment.

File: src/stories/components/MovieGrid.stories.tsx (hunk 12 above)
Comment (PetkevichVladislav) [reply src/stories/components/MovieGrid.stories.tsx:16]: fixed, thank you for your comment.

File: src/stories/components/Menu.stories.tsx (hunk 11 above)
Comment (PetkevichVladislav) [reply src/stories/components/Menu.stories.tsx:17]: fixed, thank you for your comment.

**Pull request merged**